intermediate_source/model_parallel_tutorial.py

```diff
@@ -23,11 +23,22 @@
 of model parallel. It is up to the readers to apply the ideas to real-world
 applications.
 
-**Recommended Reading:**
-
--  https://pytorch.org/ For installation instructions
--  :doc:`/beginner/blitz/data_parallel_tutorial` Single-Machine Data Parallel
--  :doc:`/intermediate/ddp_tutorial` Combine Distributed Data Parallel and Model Parallel
+**Note**: you may be wondering when you should use `DataParallel <https://pytorch.org/docs/stable/nn.html#torch.nn.DataParallel>`_
```

Comment:
Both `DataParallel` and `DistributedDataParallel` implement data parallel training. The difference is that `DataParallel` is single-process multi-thread, and only works on a single machine, while `DistributedDataParallel` is multi-process and works for both single- and multi- machine training. Even for single machine training `DistributedDataParallel` is expected to be faster than `DataParallel`.

So, the text here will need some change, e.g., sth like:

you may be wondering when you should use model parallel vs data parallel.

- If your **model** is too large to fit on one GPU, use model parallel.
- If your **data** is too large and too slow to train, use data parallel (`DataParallel` or `DistributedDataParallel`).
- If both your **model** and your **data** are too large, combine model parallel with `DistributedDataParallel`. More specifically, each `DistributedDataParallel` process uses model parallel, and all processes collectively use data parallel.